Sources/_NIODataStructures/Heap.swift

```diff
@@ -115,6 +115,21 @@ internal struct Heap<Element: Comparable> {
             return false
         }
     }
+    
+    @inlinable
+    internal mutating func removeAll(where shouldBeRemoved: (Element) throws -> Bool) rethrows {
+        guard self.storage.count > 0 else {
+            return
+        }
+        
+        for (index, element) in self.storage.enumerated().reversed() {
```

Comment:
Be very cautious here: `enumerated` does _not_ give you indices, it gives you a counter. Additionally, `EnumeratedSequence` is not a `BidirectionalCollection`, so I suspect this function call allocates, as `reversed` must iterate the entire sequence in order to produce the backwards version.

Reply:
I used `enumerated ` and `reversed` intentionally here since I am modifying the collection while iterating over it. However, your comment above is completely right that we only need a `removeFirst` or a `firstIndex(where)`. So I can remove this code anyhow and will update the PR 👍